src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -523,6 +530,18 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     if (typeof newChild === 'object' && newChild !== null) {
       switch (newChild.$$typeof) {
         case REACT_ELEMENT_TYPE: {
+          if (newChild.type === REACT_FRAGMENT_TYPE) {
+            const created = createFiberFromFragment(
+              isArray(newChild.props.children)
```

Comment:
What if it is `<>{new Set()}</>`? That is an iterable item. In all other cases arrays and iterables are treated the same. I believe this changes the semantics because you nest it one extra time.

Who do you need to do this at all? If the children set of a fragment is a single item, it should fall into the branch that optimizes for reconciling a single child. The is same as passing a single child to a Host component.

So I don't think you should need to wrapper at all.

Reply:
@sebmarkbage I'm having a tough time with this part. From my understanding, there's only a branch that optimizes for reconciling a single child in `reconcileSingleElement`. However, if we want to preserve the semantics of not having a fragment fiber in a top level array, then we need to reconcile using `reconcileChildrenArray`, which only takes in an array of children.

Reply:
Actually, I can probably recurse on `reconcileChildFibers`. Does that make sense? I'll test that right now.

Reply:
You can't recurse on reconcileChildFibers because that will flatten a nested set of single element fragments which is not what we want.

You can however fork it and if it's another fragment go for the multi-child case.

Then you can try to unify them and share code.